### PR TITLE
feat: password requirements - api 

### PR DESCRIPTION
**Pull request by kimskimchi**
## Summary

This implements the api check based on the saved settings in the db. 
+ re-implements saveSettings (synced with @dnephin; useful for test, and need it in the near future)

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Implements #2024

### CODE REVIEW

File: internal/access/credential.go [hunk 1]
```diff
@@ -94,3 +102,55 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
...
+
+	if !hasMinimumCount(settings.NumberMin, password, unicode.IsDigit) {
+		errs = append(errs, fmt.Sprintf("needs minimum %d numbers", settings.NumberMin))
+	}
+
+	if !hasMinimumCount(settings.SymbolMin, password, isValidSymbol) {
+		errs = append(errs, fmt.Sprintf("needs minimum %d symbols", settings.SymbolMin))
+	}
+
+	if len(password) > settings.LengthMin {
```
Comment (BruceMacD) [inline internal/access/credential.go:145]:
```suggestion
	if len(password) < settings.LengthMin {
```

File: internal/access/credential_test.go [hunk 2]
```diff
@@ -56,3 +57,212 @@ func TestUpdateCredentials(t *testing.T) {
...
+
+	result = hasMinimumCount(2, `[[`, isValidSymbol)
+	assert.Equal(t, result, true)
+
+	result = hasMinimumCount(2, `]]`, isValidSymbol)
+	assert.Equal(t, result, true)
+
+	result = hasMinimumCount(2, `@$%@#ss`, isValidSymbol)
+	assert.Equal(t, result, true)
+}
```
Comment (BruceMacD) [inline internal/access/credential_test.go:268]: A test for min length would be nice too

File: internal/access/credential.go (hunk 1 above)
Comment (kimskimchi) [reply internal/access/credential.go:145]: clearly missing some tests here 😂 . Pushing them soon

File: internal/access/credential.go [hunk 3]
```diff
@@ -94,3 +102,55 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
...
+		errs = append(errs, fmt.Sprintf("needs min length of %d", settings.LengthMin))
+	}
+
+	if len(errs) > 0 {
+		err := fmt.Errorf(fmt.Sprintf("%v", errs))
+		// Wrap so it is easier to parse the list of errors
+		return fmt.Errorf("cannot update password: new password does not pass requirements: %w", err)
```
[the remark this answers is not in the document]
Comment (kimskimchi) [reply internal/access/credential.go:152]: agreed

File: internal/access/credential_test.go (hunk 2 above)
Comment (kimskimchi) [reply internal/access/credential_test.go:317]: done

**Review by BruceMacD — APPROVED** [review]
🚢

**Pull request merged**